Correctly extract token from netrc file and command line

anton-petrov opened the pull request:
Resolves #156 

This fix will allow the token to load correctly from netrc file. Also, `modis_download.py` was corrected and allows to use `-T` or `--token` option for token case from command line.

For example:
```
    machine urs.earthdata.nasa.gov
    login token
    password YOURTOKEN
```

Thus, if you want to use a token instead of a username and password, you should use the word **token** as the username and the token value as a password.

veroandreo commented:
Hello @anton-petrov, thanks for the quick fix! Is the token now the only way to access the data? Since I have my .netrc file with user and password and that did not work... The only change I should make then is in the .netrc? Or is anything to be changed in the usage of modis_download.py in the command line?

veroandreo commented:
I changed my .netrc to use token and added my token there. I understand it should be read automatically, right? Now, using the same command as reported in #156, I get:

```
modis_download.py -s MOTA -p MCD12Q1.051 -t h13v11 -f 2003-01-01 -e 2001-01-01 .
Traceback (most recent call last):
  File "/usr/local/bin/modis_download.py", line 4, in <module>
    __import__('pkg_resources').run_script('pyModis==2.4.0', 'modis_download.py')
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 672, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 1479, in run_script
    exec(script_code, namespace, namespace)
  File "/usr/local/lib/python3.11/site-packages/pyModis-2.4.0-py3.11.egg/EGG-INFO/scripts/modis_download.py", line 167, in <module>
  File "/usr/local/lib/python3.11/site-packages/pyModis-2.4.0-py3.11.egg/EGG-INFO/scripts/modis_download.py", line 147, in main
  File "/usr/local/lib/python3.11/site-packages/pyModis-2.4.0-py3.11.egg/pymodis/downmodis.py", line 268, in __init__
OSError: You must provide either a token or a user and password
```

Which is the proper usage?

veroandreo commented:
If I paste the full token on the command line it does indeed work, but it looks terrible... It should read it from the .netrc, no?

anton-petrov commented:
> If I paste the full token on the command line it does indeed work, but it looks terrible... It should read it from the .netrc, no?

Yes, it should read from .netrc

anton-petrov commented:
> I changed my .netrc to use token and added my token there. I understand it should be read automatically, right? That's the point of using such a file, not writing it in the terminal. Permissions of the file are as indicated in #137.
> 
> Now, using the same command as reported in #156, I get:
> 
> ```
> modis_download.py -s MOTA -p MCD12Q1.061 -t h13v11 -f 2003-01-01 -e 2001-01-01 .
> Traceback (most recent call last):
>   File "/usr/local/bin/modis_download.py", line 4, in <module>
>     __import__('pkg_resources').run_script('pyModis==2.4.0', 'modis_download.py')
>   File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 672, in run_script
>     self.require(requires)[0].run_script(script_name, ns)
>   File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 1479, in run_script
>     exec(script_code, namespace, namespace)
>   File "/usr/local/lib/python3.11/site-packages/pyModis-2.4.0-py3.11.egg/EGG-INFO/scripts/modis_download.py", line 167, in <module>
>   File "/usr/local/lib/python3.11/site-packages/pyModis-2.4.0-py3.11.egg/EGG-INFO/scripts/modis_download.py", line 147, in main
>   File "/usr/local/lib/python3.11/site-packages/pyModis-2.4.0-py3.11.egg/pymodis/downmodis.py", line 268, in __init__
> OSError: You must provide either a token or a user and password
> ```
> 
> Which is the proper usage? It seems the .netrc is not being read neither with user and password, nor with the token.

As far as I can see, you are using version 2.4.0, the old not updated version. New fixed version must be pushed to the PyPi. Until this happens, install it from my repostiory:
```
pip install git+https://github.com/anton-petrov/pyModis.git
```

anton-petrov commented:
> Is the token now the only way to access the data?
As far as I know - it's the only approach.

veroandreo commented:
> You must provide either a token or a user and password

Yes

> > If I paste the full token on the command line it does indeed work, but it looks terrible... It should read it from the .netrc, no?
> 
> @veroandreo Yes, it should read from .netrc. Did you try my example for netrc?
> 
> ```
>     machine urs.earthdata.nasa.gov
>     login token
>     password YOURTOKEN
> ```

Yes, it didn't work. I got the error above. 

I did a `git pull` after this PR was merged and then I ran `python setup.py install` as root. So, the changes are in. Indeed,  running `modis_download.py -h` now works (it didn't before this change).

veroandreo commented:
> > Is the token now the only way to access the data?
> 
> As far as I know - it's the only approach.

Shouldn't this error: `OSError: You must provide either a token or a user and password` be changed then?

anton-petrov commented:
Personally, I'm not using netrc, this fix is just for you ;)
